### PR TITLE
feat: add manual augment input with card-based picker

### LINKED ISSUE #9: Manual augment input with state tracking
## Parent

PRD: `plans/prd.md`

## What to build

The Riot Live Client Data API does not expose augment data. Add a manual input mechanism so the player can record which augments they've selected. The Game State Manager should accept this input and incorporate it into the tracked game state, so that augment data is available for context assembly and recommendations.

The desktop shell should provide a UI for entering augment selections and display accumulated augments as part of the game state visualization.

## Acceptance criteria

- [ ] Manual augment input accepted via the desktop shell UI
- [ ] Selected augments reflected in the Game State Manager's tracked state
- [ ] Augment accumulation tracked across multiple selections within a game
- [ ] Augment data persisted to the game session in SQLite (if #7 is complete)
- [ ] Desktop shell shows current augments alongside other game state
- [ ] TDD coverage for augment accumulation and state tracking

## Blocked by

- Blocked by #8

## User stories addressed

- User story 2 (auto-detect game state — filling gaps the API doesn't expose)
- User story 3 (contextual augment recommendations — requires knowing current augments)

### PR DESCRIPTION
**Pull request by niftymonkey**
## Summary

Closes #9. Adds a card-based augment selection system for manually recording augment picks during ARAM Mayhem games.

- **AugmentCard**: Reusable card component matching in-game augment card design — icon, name, description, set badges at bottom, thick tier-colored borders (prismatic rainbow gradient, gold, silver), 2:3 aspect ratio, gradient background
- **AugmentPicker**: Searchable overlay grid of augment cards with mode/tier/set filtering, excludes already-selected augments
- **AugmentSlots**: 4-slot UI (levels 1, 7, 11, 15) — tap an empty slot to open the picker overlay, tap to choose, hover the last filled slot to see "Remove" and tap to undo the last pick. Clear button resets all.
- **Augment availability detection**: `checkAugmentAvailability()` compares player level against mode-specific thresholds to show a pulsing green "Augment available" indicator on the pending slot. Thresholds are defined per mode via `augmentSelectionLevels` on the `GameMode` interface.
- **Augment selection state**: `useAugmentSelection` hook manages picks with `useEffect`-based reset keyed on `status:gameMode`, feeds into `useEffectiveGameState` which applies selections to the active player's ModeContext with set progress recomputation
- **augment-selection.ts**: Pure functions for `addSelectedAugment` (immutable context update) and `computeSetProgress` (derives set progress from selected augments against set definitions)
- **Augments tab**: Now uses the same card grid with mode/tier/set filters plus search
- **Technical reference**: Updated with augment selection timing (levels 1, 7, 11, 15), Nexus visit requirement, and Transmute-style double augment edge case (max 5 augments, 4 chosen + 1 granted)

169 tests across 18 files, 21 new tests for augment selection, set progress, and availability detection.

### CODE REVIEW

File: src/components/AugmentCard.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      {!compact && augment.sets.length > 0 && (
+        <div className="augment-card-sets">
+          {augment.sets.map((s) => (
+            <span key={s} className="augment-card-set">
+              {s}
+            </span>
+          ))}
+        </div>
+      )}
+    </Tag>
```
[the remark this answers is not in the document]
Comment (niftymonkey) [reply src/components/AugmentCard.tsx]: Fixed — replaced all `<p>` and `<div>` inside the card with `<span>` elements (with `display: block` in CSS) so the markup is valid when rendered as a `<button>`.

File: src/components/AugmentPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  return (
+    <div className="augment-picker">
+      <input
+        type="text"
+        placeholder="Search augments..."
+        value={query}
+        onChange={(e) => setQuery(e.target.value)}
+        className="search-input"
+      />
```
[the remark this answers is not in the document]
Comment (niftymonkey) [reply src/components/AugmentPicker.tsx]: Fixed — added `aria-label="Search augments"` to the picker's search input.

File: src/hooks/useAugmentSelection.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  const [selectedAugments, setSelectedAugments] = useState<Augment[]>([]);
+  const lastGameModeRef = useRef(gameMode);
+
+  // Reset selections when the game mode changes (new game)
+  if (gameMode !== lastGameModeRef.current) {
+    lastGameModeRef.current = gameMode;
+    if (selectedAugments.length > 0) {
+      setSelectedAugments([]);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (niftymonkey) [reply src/hooks/useAugmentSelection.ts]: Fixed — rewrote the hook to use `useEffect` for reset (no synchronous state update during render) and changed the parameter to a `resetKey` string. The call site now passes `\`${gameState.status}:${gameState.gameMode}\`` so selections reset on both game mode changes and status transitions (disconnected → connected).

File: src/components/AugmentSlots.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+              className={`augment-slot augment-slot-empty${isClickable ? " augment-slot-clickable" : ""}`}
+              onClick={isClickable ? () => setPickerOpen(true) : undefined}
+            >
+              <div className="augment-slot-placeholder">
+                <span className="augment-slot-number">{i + 1}</span>
+                {isClickable && (
+                  <span className="augment-slot-hint">Tap to choose</span>
+                )}
+              </div>
+            </div>
```
[the remark this answers is not in the document]
Comment (niftymonkey) [reply src/components/AugmentSlots.tsx]: Fixed — added `role="button"`, `tabIndex={0}`, and `onKeyDown` handler (Enter/Space) to clickable empty slots for keyboard accessibility.
